cloudevents/formats/cloudevents-compact.avsc

```diff
@@ -0,0 +1,82 @@
+{
+  "namespace": "io.cloudevents.v1.avro.compact",
+  "type": "record",
+  "name": "CloudEvent",
+  "version": "1.0",
+  "doc": "Avro Compact Event Format for CloudEvents",
+  "fields": [
+    {
+      "name": "id",
+      "type": "string"
+    },
+    {
+      "name": "source",
+      "type": "string"
+    },
+    {
+      "name": "type",
+      "type": "string"
+    },
+    {
+      "name": "datacontenttype",
+      "type": [
+        "null",
+        "string"
+      ],
+      "default": null
+    },
+    {
+      "name": "dataschema",
+      "type": [
+        "null",
+        "string"
+      ],
+      "default": null
+    },
+    {
+      "name": "subject",
+      "type": [
+        "null",
+        "string"
+      ],
+      "default": null
+    },
+    {
+      "name": "time",
+      "type": [
+        "null",
+        {
+          "type": "long",
+          "logicalType": "timestamp-micros"
+        }
+      ],
+      "default": null
+    },
+    {
+      "name": "attributes",
+      "type": {
+        "type": "map",
+        "values": [
+          "null",
```

Comment:
I hadn't thought about this before, but what's the purpose of having null as a valid extension attribute value here? It makes some sense for the optional core attributes, but I don't believe CloudEvents have the concept of "here's an extension attribute, but it has no value". (The JSON format explicitly states that null values are discarded.)

Reply:
I *think* I agree with @jskeet here.. The map as-a-whole is optional, but if it's present then each entry must have a valid value.

Reply:
Removed “null”